### DISCUSSION TITLE
how to add middleware

### DISCUSSION THREAD

**Discussion by nikhil-varma-mudunuri** [pocketbase/pocketbase#4047] [Q&A] [answered]
i have a url as /matches/:matchId/:userId , is there any middleware for this url, which can we only accessed by userId same as url

**Comment by ganigeorgiev** [comment, marked as answer]
If you use the JS pb_hooks, then [`$apis.requireAdminOrOwnerAuth("userId")`](https://pocketbase.io/jsvm/functions/_apis.requireAdminOrOwnerAuth.html) should work for your use case.

Alternatively, depending on what `:matchId` is you can just perform the check inside the route (see [Retrieving current auth state](https://pocketbase.io/docs/js-routing/#retrieving-the-current-auth-state)) or [create a middleware on your own](https://pocketbase.io/docs/js-routing/#custom-middlewares).
